generate-assets/src/bin/generate.rs

```diff
@@ -48,12 +48,67 @@ fn main() -> anyhow::Result<()> {
         },
     )?;
 
+    let latest_bevy_version = get_latest_bevy_version(&db)?;
+
+    sort_section(&mut asset_root_section.content, &latest_bevy_version);
+
     asset_root_section
         .write(Path::new(&content_dir), Path::new(""), 0)
         .expect("Failed to write assets section");
     Ok(())
 }
 
+/// Sort the assets in the section so that:
+/// - Assets that have been manually assigned an order in `bevy-assets` are first
+/// - Assets that are semver compatible with Bevy are next
+/// - If all else is equal, sort randomly
+fn sort_section(nodes: &mut [AssetNode], latest_bevy_version: &semver::Version) {
+    for node in nodes.iter_mut() {
+        if let AssetNode::Section(ref mut section) = node {
+            sort_section(&mut section.content, latest_bevy_version);
+        }
+    }
+
+    let mut to_sort = vec![];
+    for node in nodes.iter_mut() {
+        let is_semver_compat = node_semver_compat_with(node, latest_bevy_version);
+        let random: u32 = rand::random();
```

Comment:
Would move this line down to where it's used, since we might `continue` before needing to generate a random number.